ext/phar/util.c

```diff
@@ -1402,6 +1402,14 @@ static int phar_call_openssl_signverify(int is_sign, php_stream *fp, zend_off_t
 	} else {
 		ZVAL_EMPTY_STRING(&zp[0]);
 	}
+	if (sig_type == PHAR_SIG_OPENSSL_512) {
+		ZVAL_LONG(&zp[3], 9); /* value from openssl.c #define OPENSSL_ALGO_SHA512 9 */
+	} else if (sig_type == PHAR_SIG_OPENSSL_256) {
+		ZVAL_LONG(&zp[3], 7); /* value from openssl.c #define OPENSSL_ALGO_SHA256 7 */
+	} else {
+		/* don't rely on default value which may change in the future */
+		ZVAL_LONG(&zp[3], 1); /* value from openssl.c #define OPENSSL_ALGO_SHA1   1 */
+	}
```

Comment:
Maybe leave a comment in openssl.c that phar/util.c needs to be updated if constant values change.

Reply:
done